### PR TITLE
Task/ob 105 UI login

### PR DESCRIPTION
**Pull request by ps-eqsdigital**
ui for user login

### CODE REVIEW

File: api/Business/BusinessObjects/UserBusinessObject.cs [hunk 1]
```diff
@@ -57,15 +72,17 @@ public async Task<OperationResult> Update(Guid uuid, User record)
...
         {
             return await ExecuteOperation(async () => {
 
                 if (sort !=0 && sort != 1)
                 {
                     throw new Exception();
                 }
-                List<User> result =await _userDataAccessObject.FilterUsers(search,sort);
+                List<User> users =await _userDataAccessObject.FilterUsers(search,sort);
+                var result = users.Select(u => new UserBusinessModel(u)).ToList();
```
Comment (kripton69) [inline api/Business/BusinessObjects/UserBusinessObject.cs]: type

File: api/Business/BusinessObjects/UserBusinessObject.cs [hunk 2]
```diff
@@ -89,12 +113,72 @@ public async Task<OperationResult<CreateUserBusinessModel>> Insert(User record)
...
+                    } 
+                };
+            });
+        }
+
+        public async Task<OperationResult> Logout(string token)
+        {
+            return await ExecuteOperation(async () =>
+            {
+                var result = await _userDataAccessObject.GetTokenUuidByToken(token);
```
Comment (kripton69) [inline api/Business/BusinessObjects/UserBusinessObject.cs]: type

File: api/Business/BusinessObjects/UserBusinessObject.cs [hunk 3]
```diff
@@ -119,5 +203,27 @@ static bool IsValidEmail(string email)
...
+            .SetBasePath(Directory.GetCurrentDirectory())
+            .AddJsonFile("appsettings.json")
+            .Build();
+
+            List<Claim> claims = new List<Claim>
+            {
+                new Claim(ClaimTypes.Name, user.Name!),
+            };
+
+            var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(configuration.GetSection("AppSettings:Token").Value!));
```
Comment (kripton69) [inline api/Business/BusinessObjects/UserBusinessObject.cs]: types

File: api/api/Middleware/TokenValidationMiddleware.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            // Create a scope to access scoped services
+            using (var scope = context.RequestServices.CreateScope())
+            {
+                var serviceProvider = scope.ServiceProvider;
```
Comment (kripton69) [inline api/api/Middleware/TokenValidationMiddleware.cs]: types

File: api/Business/BusinessObjects/UserBusinessObject.cs (hunk 1 above)
Comment (ps-eqsdigital) [reply api/Business/BusinessObjects/UserBusinessObject.cs]: done

File: api/Business/BusinessObjects/UserBusinessObject.cs (hunk 2 above)
Comment (ps-eqsdigital) [reply api/Business/BusinessObjects/UserBusinessObject.cs]: done

File: api/Business/BusinessObjects/UserBusinessObject.cs (hunk 3 above)
Comment (ps-eqsdigital) [reply api/Business/BusinessObjects/UserBusinessObject.cs]: done

File: api/api/Middleware/TokenValidationMiddleware.cs (hunk 4 above)
Comment (ps-eqsdigital) [reply api/api/Middleware/TokenValidationMiddleware.cs]: done

File: api/Business/BusinessModels/UserBusinessModel.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace Business.BusinessModels
+{
+    public class UserBusinessModel
+    {
+        public Guid Uuid { get; set; }
+        public string? Name { get; set; }
+        public string? Email { get; set; }
+        public string? Picture {  get; set; }
+        public UserBusinessModel() { }
```
Comment (tentone) [inline api/Business/BusinessModels/UserBusinessModel.cs]: Docs for all these :)

File: api/api/Controllers/UserController.cs [hunk 6]
```diff
@@ -66,5 +69,27 @@ public async Task<ActionResult> ListFilteredUsers(string search, int sort)
...
+        [HttpPost("logout"), Authorize]
+        public async Task<ActionResult> Logout(string token)
+        {
+            OperationResult result = await _userBusinessObject.Logout(token);
+            if (result.Exception is Exception)
+            {
+                return StatusCode(400);
+            }
+            return StatusCode(200);
+        }
```
Comment (tentone) [inline api/api/Controllers/UserController.cs]: Use status codes from HttpStatusCode enum.

https://learn.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?view=net-7.0

**Review by tentone — APPROVED** [review]

**Pull request merged**